pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -309,8 +300,8 @@ func TestCollectObservationLog(t *testing.T) {
 	for name, test := range testCases {
 		t.Run(name, func(t *testing.T) {
 			actual, err := CollectObservationLog(test.filePath, test.metrics, test.filters, test.fileFormat)
```

Comment:
```suggestion
	testMetricsFileName := "test-metrics"
	testCases := map[string]struct {
		testData string
		metrics    []string
		...
	}{
		"": {
			...
		},
		...
	}
	for name, test := range testCases {
		t.Run(name, func(t *testing.T) {
			tmpDir := t.TempDir()
			if len(test.testData) != 0 {
				if err := os.WriteFile(filepath.Join(tmpDir, testMetricsFileName), []byte(test.testData), 0600); err != nil {
					return t.Fatal(err)
				}
			}
			actual, err := CollectObservationLog(test.filePath, test.metrics, test.filters, test.fileFormat)
```

Once we introduce the `tmp` dir, the test data should be generated more declaratively.

Reply:
Okay, I'll refactor the current code.

Reply:
I assumed the same as the currently one. (decription)

Reply:
Okay, I see.